### PR TITLE
Add public IP detection (from network interfaces), node outputs a war…

### PR DESCRIPTION
**Pull request by adagys**
…ning if no public IP detected or specified in the config.

The warning is being outputted to both the logfile and stdout:
```
   ______               __
  / ____/     _________/ /___ _
 / /     __  / ___/ __  / __ `/         "It's OK computer, I go to sleep after
/ /___  /_/ / /  / /_/ / /_/ /          twenty minutes of inactivity too!"
\____/     /_/   \__,_/\__,_/

--- MILESTONE 9 -------------------------------------------------------------------

Logs can be found in                    : /.../logs
Database connection url is              : jdbc:h2:tcp://x.x.x.x:50497/node
WARNING: The specified messaging host "localhost" is private, this node will not be reachable by any other nodes outside the private network.
Node listening on address               : localhost:10002
Loaded plugins                          : net.corda.attachmentdemo.plugin.AttachmentDemoPlugin
Node started up and registered in 23.23 sec
```

### CODE REVIEW

File: node/src/test/kotlin/net/corda/node/utilities/AddressUtilsTests.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package net.corda.node.utilities
+
+import org.junit.Test
+import java.net.InetAddress
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class AddressUtilsTests {
+    @Test
+    fun `correctly determines if the provided address is public`() {
```
Comment (adagys) [inline node/src/test/kotlin/net/corda/node/utilities/AddressUtilsTests.kt]: do we care about IP6?

Comment (mikehearn) [inline node/src/test/kotlin/net/corda/node/utilities/AddressUtilsTests.kt]: Yes, but IPv6 addresses are always (touch wood) global. I'm sure someone will figure out a way to make them not work properly but I've yet to see a machine with a non-global IPv6 address, and all my devices+servers are IPv6.

**Comment by mikehearn** [comment]
Looks good. Just add a few IPv6 test cases. By the way, if you do log.error() then it'll be printed to the console as well anyway, so no need for an additional println.

**Comment by mikehearn** [comment]
Also did you try it out on an azure box?

**Comment by adagys** [comment]
Yeah I tried int on a Debian VM. Also my mac, an Ubuntu & Windows machines.

**Comment by adagys** [comment]
I don't really want to output an error, as it's not really an error if someone's running a local (demos) or a private network.

**Pull request merged**